### ISSUE TITLE
request: prevent re-linting when re-opening unmodified buffer

### ISSUE THREAD

**Issue by unphased** [w0rp/ale#1418]
When linters are very expensive (such as `clang-tidy`) it places a penalty on using vim by opening buffers without using windows and switching back and forth between them. With this, ALE will re-lint each buffer when re-entering it. When the buffers are opened as windows side by side or as tabs, this does not occur. 

I think that a reasonably practical workaround might be to `let g:ale_lint_on_enter = 0` but my thought was this could be made a tiny bit more intelligent somehow.

**Comment by w0rp** [comment]
That's how it's supposed to work. What might make sense is adding a new setting for only checking files when they changed outside of Vim, which would default to `g:ale_lint_on_enter`, so it won't suddenly turn on for users who have that option turned off already.

**Comment by w0rp** [comment]
Another option for checking files which are opened for the first time might make sense, so you can only check new buffers, buffers changed outside of Vim, or buffers you saved yourself.

**Comment by w0rp** [comment]
Actually, it would make more sense to support string values for `lint_on_enter`, like ALE does for `g:ale_lint_on_text_changed`.

**Comment by unphased** [comment]
Yeah I'd agree that this only makes sense in the context of the current lint_on_enter being enabled, and so the 3 mode configuration would then make more sense. 

Could it be as simple as changing the Autocmd that linting is mapped to?

**Comment by w0rp** [comment]
There are three autocmd events, so all we'd need to do is select which ones to run, based on the option.

**Comment by unphased** [comment]
Nice, I'll look into implementing this when I can. thanks.
